pkg/controller/argocd/service.go

```diff
@@ -393,7 +393,17 @@ func (r *ReconcileArgoCD) reconcileServerMetricsService(cr *argoprojv1a1.ArgoCD)
 // reconcileServerService will ensure that the Service is present for the Argo CD server component.
 func (r *ReconcileArgoCD) reconcileServerService(cr *argoprojv1a1.ArgoCD) error {
 	svc := newServiceWithSuffix("server", "server", cr)
-	if argoutil.IsObjectFound(r.client, cr.Namespace, svc.Name, svc) {
+
+	if err := applyReconcilerHook(cr, svc, ""); err != nil {
+		return err
+	}
+
+	existingSvc := &corev1.Service{}
```

Comment:
@chetan-rns After the cluster generates a secret for your service, your PodSpec should mount it, and the Pod will run after it becomes available.

Did you verify that the argocd-server has started with the certificate that the ingress-operator has created for you ? I dont see the changes for that in this PR.

Reply:
@iam-veeramalla We're trying to mount the generated secrets in a pod created by the downstream operator. Hence there are no mounts in the argocd-server cc @shubhamagarwal19

Reply:
@chetan-rns @shubhamagarwal19  any reason why you dont want to implement that here ?

Reply:
The pod where we want to mount the secrets is created by the downstream operator.

Reply:
as discussed we would need to mount the secret to enable TLS configuration for Argo CD Server